riscv-demo/riscv_demo/soc.py

```diff
@@ -7,12 +7,89 @@
 from amaranth_soc.wishbone.sram import WishboneSRAM
 
 from minerva.core import Minerva
+from chipflow_lib.platforms.iostream import IOShape
 
 from .ips.qspi import QSPIController, WishboneQSPIFlashController
 from .ips.uart import UARTPhy, UARTPeripheral
+from pprint import pprint, pformat
 
 
 __all__ = ["DemoSoC"]
 
 
-# TODO
+class DemoSoC(wiring.Component):
+    def pins():
+        return IOShape({
+                'qspi': QSPIController.pins(),
```

Comment:
how would this look if I want to expose a single pin directly from DemoSoC (like, say, a fault indicator or specific interrupt or something like that)